api/tests/test_depth_check.py

```diff
@@ -0,0 +1,207 @@
+    def test_valid_depth_query(self):
+        backend = DepthAnalysisBackend()
+        client = Client(schema)
+        query = client.execute(
+            '''
+            {
+                getSectorById(id: 1) {
+                    groups {
+                        edges {
+                            node {
+                                sectorId
+                            }
+                        }
+                    }
+                }
+            }
+            ''', backend=backend)
+        result_refr = {
+            "data": {
+                "getSectorById": [
+                    {
+                        "groups": {
+                            "edges": [
+                                {
+                                    "node": {
+                                        "sectorId": 1
+                                    }
+                                }
+                            ]
+                        }
+                    }
+                ]
+            }
+        }
+        self.assertDictEqual(result_refr, query)
+
+    def test_invalid_depth_query(self):
+        backend = DepthAnalysisBackend()
+        client = Client(schema)
+        executed = client.execute(
+            '''
+            {
+              getSectorById(id: 1) {
+                groups{
+                  edges{
+                    node{
+                      groupSector{
+                        groups{
+                          edges{
+                            node{
+                              groupSector{
+                                groups{
+                                  edges{
+                                    node{
+                                      groupSector{
+                                        groups{
+                                          edges{
+                                            node{
+                                              groupSector{
+                                                groups{
+                                                  edges{
+                                                    node{
+                                                      groupSector
+                                                    }
+                                                  }
+                                                }
+                                              }
+                                            }
+                                          }
+                                        }
+                                      }
+                                    }
+                                  }
+                                }
+                              }
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+            ''', backend=backend)
+        result_refr = {
```

Comment:
This isn't being used so we should get rid of it...

Reply:
This is needed because the back end isn't apart of the schema so it has no way of knowing it is there. The back end is added in when we create the url rule

Reply:
Sadly that is how flask-graphql is setup